sdk/servicebus/azure_messaging_servicebus/src/client.rs

```diff
@@ -0,0 +1,559 @@
+    }
+    /// Returns a builder which can be used to create a new instance of [`ServiceBusClient`].
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use azure_messaging_servicebus::ServiceBusClient;
+    /// use azure_identity::DeveloperToolsCredential;
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    ///     let credential = DeveloperToolsCredential::new(None)?;
+    ///     let client = ServiceBusClient::builder()
+    ///         .open("my-servicebus.servicebus.windows.net", credential.clone()).await?;
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn builder() -> ServiceBusClientBuilder {
+        ServiceBusClientBuilder::new()
+    }
+
+    /// Creates a new client internally using the provided options.
+    pub(crate) async fn new_internal(
+        fully_qualified_namespace: &str,
+        credential: Arc<dyn TokenCredential>,
+        options: Option<ServiceBusClientOptions>,
+    ) -> Result<Self> {
+        let endpoint = format!("amqps://{}:5671", fully_qualified_namespace);
+        let endpoint_url = Url::parse(&endpoint).map_err(|e| {
+            ServiceBusError::new(
+                ErrorKind::InvalidRequest,
+                format!("Invalid endpoint URL: {}", e),
+            )
+        })?;
+        let namespace = fully_qualified_namespace.to_string();
+
+        let connection = Arc::new(AmqpConnection::new());
+
+        // Create authorizer with the credential for token-based authentication
+        let authorizer = Arc::new(Authorizer::new(Arc::downgrade(&connection), credential));
+
+        let connection_options = Self::build_connection_options(options.clone());
+        connection
+            .open(
+                "servicebus-client".to_string(),
+                endpoint_url,
+                connection_options,
+            )
+            .await?;
+
+        Ok(Self {
+            connection,
+            namespace,
+            options: options.unwrap_or_default(),
+            authorizer: Some(authorizer),
+        })
+    }
+
+    /// Creates a sender for the specified queue or topic.
+    pub async fn create_sender(
+        &self,
+        queue_or_topic_name: &str,
+        _options: Option<CreateSenderOptions>,
+    ) -> Result<Sender> {
+        // Authorize the path if we have a credential-based client
+        self.authorize_path(queue_or_topic_name).await?;
+
+        Sender::new(
+            self.connection.clone(),
+            queue_or_topic_name.to_string(),
+            self.options.clone(),
+        )
+        .await
+    }
+
+    /// Creates a receiver for the specified queue with options.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use azure_identity::DeveloperToolsCredential;
+    /// use azure_messaging_servicebus::{ServiceBusClient, CreateReceiverOptions, ReceiveMode, SubQueue};
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Nit: (and to @LarryOsterman as well) it's more common not to show boilerplate code like `main`. Put `#` before these two lines and the closing `Ok(())` and `#`, then delete the tabs in the example. Basically, just show the imports that are necessary for the rendered code and the code that is important. Forget all the fluff. e.g.

```rust
/// # Examples
///
/// ```no_run
/// import foo;
/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
/// foo::do_async().await
/// # Ok(()) }
```

Reply:
If you use something like `async fn example()` you don't need to do `#[tokio::main] async fn main()` - especially if it's `no_run`, `main` just isn't necessary.